File: BOJ/Graph/Silver/1991/1991.py
n = int(input())
a = ord("A")
arr = [(0,0) for _ in range(n)]
first = ""
middle = ""
last = ""
def dfs(cur) :
    global first,middle,last
    if cur == -19 : return
    first+=chr(cur+a)
    dfs(arr[cur][0])
    middle+=chr(cur+a)
    dfs(arr[cur][1])    
    last+=chr(cur+a)

for i in range(n) :
    root,left,right = map(ord,input().split())
    arr[root-a] = (left-a,right-a)
print(arr)

dfs(0)
print(first,middle,last, sep = '\n')

File: BOJ/Graph/Silver/1991/1991.swift
let n = Int(readLine()!)!
let a = Int(UnicodeScalar("A").value)
var arr = Array(repeating: (0,0), count: n)
var first = ""
var middle = ""
var last = ""
for _ in 0..<n {
    let input = readLine()!.split(separator : " ").map{Int(UnicodeScalar(String($0))!.value)-a},(root,left,right) = (input[0],input[1],input[2])
    arr[root] = (left,right)
}
dfs(0)
print(first,middle,last, separator : "\n")

func dfs(_ cur : Int) {
    if cur == -19 {return}
    first+=String(UnicodeScalar(cur+a)!)
    dfs(arr[cur].0)
    middle+=String(UnicodeScalar(cur+a)!)
    dfs(arr[cur].1)    
    last+=String(UnicodeScalar(cur+a)!)
}